lighthouse-viewer/test/viewer-test-pptr.js

```diff
@@ -0,0 +1,124 @@
+      {category, selector}
+    );
+  }
+
+  before(async function() {
+    server.listen(portNumber, 'localhost');
+
+    // start puppeteer
+    browser = await puppeteer.launch({
+      headless: false,
+      executablePath: process.env.CHROME_PATH,
+    });
+    viewerPage = await browser.newPage();
+    viewerPage.on('pageerror', pageError => pageErrors.push(pageError));
+    await viewerPage.goto(viewerUrl, {waitUntil: 'networkidle2', timeout: 3000});
+    const fileInput = await viewerPage.$('#hidden-file-input');
+    await fileInput.uploadFile(sampleLhr);
+    await viewerPage.waitForSelector('.lh-container', {timeout: 3000});
+  });
+
+  after(async function() {
+    await Promise.all([
+      new Promise(resolve => server.close(resolve)),
+      browser && browser.close(),
+    ]);
+  });
+
+
+  const selectors = {
+    audits: '.lh-audit, .lh-timeline-metric, .lh-perf-hint',
+    titles: '.lh-score__title, .lh-perf-hint__title, .lh-timeline-metric__title',
+  };
+
+  it('should load with no errors', async () => {
+    assert.deepStrictEqual(pageErrors, []);
+  });
+
+  it('should contain all categories', async () => {
+    const categories = await viewerPage.$$(`#${lighthouseCategories.join(',#')}`);
+    assert.equal(
+      categories.length,
+      lighthouseCategories.length,
+      `${categories.join(' ')} does not match ${lighthouseCategories.join(' ')}`
+    );
+  });
+
+  it('should contain audits of all categories', async () => {
+    for (const category of lighthouseCategories) {
+      let expected = getAuditsOfCategory(category).length;
+      if (category === 'performance') {
+        expected = getAuditsOfCategory(category).filter(a => !!a.group).length;
+      }
+
+      const elementCount = await getAuditElementsCount({category, selector: selectors.audits});
+
+      assert.equal(
+        expected,
+        elementCount,
+        `${category} does not have the correct amount of audits`
+      );
+    }
+  });
+
+  it('should contain a filmstrip', async () => {
+    const filmstrip = await viewerPage.$('.lh-filmstrip');
+
+    assert.ok(!!filmstrip, `filmstrip is not available`);
+  });
+
+  it('should not have any unexpected audit errors', async () => {
+    function getDebugStrings(elems, selectors) {
+      return elems.map(el => {
+        const audit = el.closest(selectors.audits);
+        const auditTitle = audit && audit.querySelector(selectors.titles);
+        return {
+          debugString: el.textContent,
+          title: auditTitle ? auditTitle.textContent : 'Audit title unvailable',
+        };
+      });
+    }
+
+    const auditErrors = await viewerPage.$$eval('.lh-debug', getDebugStrings, selectors);
+    const errors = auditErrors.filter(item => item.debugString.includes('Audit error:'));
+    const unexpectedErrrors = errors.filter(item => {
+      return !item.debugString.includes('Required RobotsTxt gatherer did not run');
```

Comment:
can you confirm the savedartifacts are out of date?

Reply:
yeah, no `RobotsTxt` artifact in `artifacts.json`